File: javaWebApp/src/main/java/com/saeinwoojoo/java/webapp/webview/WebChromeClientEx.java
package com.saeinwoojoo.java.webapp.webview;

import android.os.Message;
import android.util.Log;
import android.webkit.JsResult;
import android.webkit.WebChromeClient;
import android.webkit.WebView;

import com.saeinwoojoo.java.webapp.BaseActivity;

/**
 * WebChromeClientEx
 *
 */
public class WebChromeClientEx extends WebChromeClient {

	private static final String LOG_TAG = "WebChromeClientEx";

	private BaseActivity mBaseActivity;

	public WebChromeClientEx(BaseActivity baseActivity) {
		super();
		mBaseActivity = baseActivity;
	}

	@Override
	public boolean onJsAlert(WebView view, String url, String message, JsResult result) {
		Log.i(LOG_TAG, "onJsAlert()...");
		result.confirm();
		return true;
	}

	@Override
	public void onProgressChanged(WebView view, int newProgress) {
		super.onProgressChanged(view, newProgress);
//		Log.d(LOG_TAG, "onProgressChanged(): " + newProgress);

		if (null != mBaseActivity)
			mBaseActivity.setTitleProgress(newProgress);
	}
}

File: kotlinWebApp/src/main/java/com/saeinwoojoo/kotlin/webapp/webview/WebChromeClientEx.kt
package com.saeinwoojoo.kotlin.webapp.webview

import android.util.Log
import android.webkit.JsResult
import android.webkit.WebChromeClient
import android.webkit.WebView
import com.saeinwoojoo.kotlin.webapp.BaseActivity

/**
 * WebChromeClientEx
 *
 */
class WebChromeClientEx(private val mBaseActivity: BaseActivity?) : WebChromeClient() {

    override fun onJsAlert(view: WebView, url: String, message: String, result: JsResult): Boolean {
        Log.i(LOG_TAG, "onJsAlert()...")
        result.confirm()
        return true
    }

    override fun onProgressChanged(view: WebView, newProgress: Int) {
        super.onProgressChanged(view, newProgress)
//        Log.d(LOG_TAG, "onProgressChanged(): $newProgress")
        mBaseActivity?.setTitleProgress(newProgress)
    }

    companion object {

        private const val LOG_TAG = "WebChromeClientEx"
    }
}